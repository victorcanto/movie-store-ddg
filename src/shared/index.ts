export * from './models/cart.interfaces';
export * from './models/favorites.interfaces';
export * from './models/movies.interfaces';
export * from './utils/store/cart.utils';
export * from './utils/store/favorites.utils';
export * from './utils/test-utils';
export * from './utils/currency.utils';
export * from './utils/states-and-cities';
