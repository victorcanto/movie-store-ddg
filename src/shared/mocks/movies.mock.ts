import { Movie } from '../models/movies.interfaces';

export const moviesMock: Movie[] = [
	{
		adult: false,
		backdrop_path: '/ugS5FVfCI3RV0ZwZtBV3HAV75OX.jpg',
		genre_ids: [16, 878, 28],
		genres: [
			{ id: 16, name: 'Animação' },
			{ id: 878, name: 'Ficção científica' },
			{ id: 28, name: 'Ação' },
		],
		id: 610150,
		original_language: 'ja',
		original_title: 'ドラゴンボール超 スーパーヒーロー',
		overview:
			'The Red Ribbon Army, an evil organization that was once destroyed by Goku in the past, has been reformed by a group of people who have created new and mightier Androids, Gamma 1 and Gamma 2, and seek vengeance against Goku and his family.',
		popularity: 16585.915,
		poster_path: '/rugyJdeoJm7cSJL1q4jBpTNbxyU.jpg',
		release_date: '2022-06-11',
		title: 'Dragon Ball Super: Super Hero',
		video: false,
		vote_average: 8,
		vote_count: 959,
		price: 79.99,
	},
	{
		adult: false,
		backdrop_path: '/vvObT0eIWGlArLQx3K5wZ0uT812.jpg',
		genre_ids: [28, 12, 14],
		genres: [
			{ id: 16, name: 'Animação' },
			{ id: 878, name: 'Ficção científica' },
			{ id: 28, name: 'Ação' },
		],
		id: 616037,
		original_language: 'en',
		original_title: 'Thor: Love and Thunder',
		overview:
			'After his retirement is interrupted by Gorr the God Butcher, a galactic killer who seeks the extinction of the gods, Thor Odinson enlists the help of King Valkyrie, Korg, and ex-girlfriend Jane Foster, who now inexplicably wields Mjolnir as the Relatively Mighty Girl Thor. Together they embark upon a harrowing cosmic adventure to uncover the mystery of the God Butcher’s vengeance and stop him before it’s too late.',
		popularity: 5372.967,
		poster_path: '/pIkRyD18kl4FhoCNQuWxWu5cBLM.jpg',
		release_date: '2022-07-06',
		title: 'Thor: Love and Thunder',
		video: false,
		vote_average: 6.8,
		vote_count: 2223,
		price: 79.99,
	},
	{
		adult: false,
		backdrop_path: '/xfNHRI2f5kHGvogxLd0C5sB90L7.jpg',
		genre_ids: [16, 28, 10751, 878, 35],
		genres: [
			{ id: 16, name: 'Animação' },
			{ id: 878, name: 'Ficção científica' },
			{ id: 28, name: 'Ação' },
			{
				id: 10751,
				name: 'Família',
			},
			{
				id: 878,
				name: 'Ficção científica',
			},
			{
				id: 35,
				name: 'Comédia',
			},
		],
		id: 539681,
		original_language: 'en',
		original_title: 'DC League of Super-Pets',
		overview:
			'When Superman and the rest of the Justice League are kidnapped, Krypto the Super-Dog must convince a rag-tag shelter pack - Ace the hound, PB the potbellied pig, Merton the turtle and Chip the squirrel - to master their own newfound powers and help him rescue the superheroes.',
		popularity: 4630.633,
		poster_path: '/r7XifzvtezNt31ypvsmb6Oqxw49.jpg',
		release_date: '2022-07-27',
		title: 'DC League of Super-Pets',
		video: false,
		vote_average: 7.6,
		vote_count: 360,
		price: 79.99,
	},
];
