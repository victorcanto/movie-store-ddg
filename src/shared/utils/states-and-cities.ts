import statesAndCities from '../data/estados-cidades.json';

export const getBrazilianStates = () =>
	statesAndCities.estados.map(({ nome }) => nome);

export const getBrazilianCitiesByState = (stateName: string) =>
	statesAndCities.estados.find(({ nome }) => nome === stateName)?.cidades;
