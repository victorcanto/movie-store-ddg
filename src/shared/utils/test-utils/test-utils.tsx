import React, { PropsWithChildren } from 'react';
import { render as rtlRender } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { setupStore, AppStore, RootState } from '../../../store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}

export function renderWithProviders(
	ui: React.ReactElement,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	}: ExtendedRenderOptions = {},
) {
	function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
		return (
			<Provider store={store}>
				<BrowserRouter>{children}</BrowserRouter>
			</Provider>
		);
	}
	return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';

export { renderWithProviders as render };
