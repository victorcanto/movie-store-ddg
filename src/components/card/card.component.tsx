import React, { useEffect, useState } from 'react';
import {
	Box,
	Card as MuiCard,
	CardActions,
	CardContent,
	CardMedia,
	Chip,
	IconButton,
	Typography,
	useTheme,
} from '@mui/material';

import { Movie } from '../../shared';
import { StarIcon } from '../star-icon/star-icon.component';
import { FavoriteIcon } from '../favorite-icon/favorite-icon.component';
import { AddToCartButton } from '../add-to-cart-button/add-to-cart-button.component';
import { Price } from '../price/price.component';

export interface CardProps {
	item: Movie;
}

export function Card({ item }: CardProps) {
	const theme = useTheme();

	return (
		<MuiCard
			data-testid='card'
			sx={{
				height: '100%',
				display: 'flex',
				flexDirection: 'column',
				justifyContent: 'space-between',
				borderRadius: theme.shape.borderRadius,
				':hover': {
					boxShadow: theme.shadows[3],
					transition: '100ms linear',
				},
			}}
		>
			<Box sx={{ position: 'relative' }}>
				<IconButton
					sx={{
						position: 'absolute',
						top: '10px',
						right: '10px',
					}}
				>
					<FavoriteIcon item={item} />
				</IconButton>

				<CardMedia
					sx={{ boxShadow: theme.shadows[3] }}
					component='img'
					width='100%'
					image={`https://image.tmdb.org/t/p/w500/${item.poster_path}`}
					alt={item.title}
				/>
			</Box>
			<CardContent
				sx={{
					display: 'flex',
					flexDirection: 'column',
					justifyContent: 'space-between',
					gap: 1,
				}}
			>
				<Box
					sx={{
						display: 'flex',
						flexDirection: 'column',
						alignItems: 'center',
						justifyContent: 'center',
					}}
				>
					<Box
						sx={{
							display: 'flex',
							justifyContent: 'space-between',
							width: '100%',
							gap: 1,
						}}
					>
						<Typography noWrap variant='h5' fontWeight={600}>
							{item.title}
						</Typography>
						<Box
							sx={{
								display: 'flex',
								alignItems: 'center',
							}}
						>
							<StarIcon />
							<Typography>{item.vote_count}</Typography>
						</Box>
					</Box>

					<Box sx={{ display: 'flex', flexWrap: 'wrap' }}>
						{item.genres.map(({ name }) => (
							<Chip
								label={name}
								sx={{
									color: theme.palette.secondary.light,
								}}
								size='small'
							/>
						))}
					</Box>

					<Typography variant='h4' textAlign='center'>
						<Price value={item.price} />
					</Typography>
				</Box>
			</CardContent>
			<CardActions sx={{ width: '100%', p: 0 }}>
				<AddToCartButton item={item} />
			</CardActions>
		</MuiCard>
	);
}
