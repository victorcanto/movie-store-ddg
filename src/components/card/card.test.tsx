import React from 'react';
import { render } from '../../shared/utils/test-utils';
import { Card } from './card.component';
import { moviesMock } from '../../shared/mocks';

test('snapshot of the base state', () => {
	const { asFragment } = render(<Card item={moviesMock[0]} />);
	expect(asFragment()).toMatchSnapshot();
});

test('should have in the document', () => {
	const { getByTestId } = render(<Card item={moviesMock[0]} />);
	expect(getByTestId('card')).toBeInTheDocument();
});
