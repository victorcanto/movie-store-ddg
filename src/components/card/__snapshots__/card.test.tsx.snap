// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the base state 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-14spjgz-MuiPaper-root-MuiCard-root"
    data-testid="card"
  >
    <div
      class="MuiBox-root css-79elbk"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1xgqqg8-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          aria-label="Favoritar"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1rr6odo-MuiSvgIcon-root"
          data-mui-internal-clone-element="true"
          data-testid="favorite-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <img
        alt="Dragon Ball Super: Super Hero"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-10q1o9j-MuiCardMedia-root"
        src="https://image.tmdb.org/t/p/w500//rugyJdeoJm7cSJL1q4jBpTNbxyU.jpg"
        width="100%"
      />
    </div>
    <div
      class="MuiCardContent-root css-wvc621-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-u6v8er"
      >
        <div
          class="MuiBox-root css-1ys39dj"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap css-16nexq1-MuiTypography-root"
          >
            Dragon Ball Super: Super Hero
          </h5>
          <div
            class="MuiBox-root css-70qvj9"
          >
            <svg
              aria-hidden="true"
              aria-label="Número de votos"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-mey93b-MuiSvgIcon-root"
              data-mui-internal-clone-element="true"
              data-testid="star-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              959
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root css-2imjyh"
        >
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-aeaiuv-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
            >
              Animação
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-aeaiuv-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
            >
              Ficção científica
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-aeaiuv-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
            >
              Ação
            </span>
          </div>
        </div>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-149xwtl-MuiTypography-root"
        >
          <span
            data-testid="price"
          >
            R$79.99
          </span>
        </h4>
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-1eu7e7c-MuiCardActions-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-rf0srl-MuiButtonBase-root-MuiButton-root"
        data-testid="add-to-cart-button"
        tabindex="0"
        to="/"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-b2zrw0-MuiTypography-root"
        >
          ADICIONAR
        </h6>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
